src/server/__tests__/autoCleanup.disabled.test.ts

```diff
@@ -1,30 +1,38 @@
 import { useEffect } from 'react'
 
-import { ReactHooksRenderer } from '../../types/react'
+import { ReactHooksServerRenderer } from '../../types/react'
 
 // This verifies that if RHTL_SKIP_AUTO_CLEANUP is set
 // then we DON'T auto-wire up the afterEach for folks
 describe('skip auto cleanup (disabled) tests', () => {
-  let cleanupCalled = false
-  let renderHook: ReactHooksRenderer['renderHook']
+  const cleanups: Record<string, boolean> = {
+    ssr: false,
+    hydrated: false
+  }
+  let renderHook: ReactHooksServerRenderer['renderHook']
```

Comment:
I've improved quite a few of the server renderer tests to actually test the behaviour pre/post hydration.  Some already did this but others not so I made them consistent.